### PR TITLE
Docs: Clarify that ignorePattern should be a string (refs #13029)

### PR DESCRIPTION
**Pull request by btmills**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

In #13029, the author based the docs on `ignorePattern` on `lines-around-comment`, which says the option should be a "regular expression". Here I update both places to clarify that the option should be a string that will be used to construct a regular expression.

### CODE REVIEW

File: docs/rules/lines-around-comment.md
Comment (mdjermanovic) [inline docs/rules/lines-around-comment.md:440]: Should we say "change", since this doesn't replace the default pattern?

Comment (btmills) [reply docs/rules/lines-around-comment.md:440]: Good point. How about this?

```suggestion
By default this rule ignores comments starting with the following words: `eslint`, `jshint`, `jslint`, `istanbul`, `global`, `exported`, `jscs`. To ignore more comments in addition to the defaults, set the `ignorePattern` option to a string pattern that will be passed to the [`RegExp` constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp).
```

**Review by mdjermanovic — APPROVED** [review]
LGTM! Waiting for CI to finish before merging.

**Pull request merged**
